### PR TITLE
Added execution model and shared objects docs into documentation

### PR DESCRIPTION
**Pull request by LiraLemur**
Signed-off-by: Sara <lira.lemur@gmail.com>

### Description of the Change

Added old documents from GitHub to documentation. 2 docs were added: https://github.com/hyperledger-archives/iroha/issues/1423
https://github.com/hyperledger-archives/iroha/issues/1498

### Benefits

We will have fuller architecture docs

### Possible Drawbacks 

Some information in the docs might be outdated. It was arranged with @lebdron that we will review and fix those parts within the readthedocs documentation.

### CODE REVIEW

File: docs/source/architecture/shared_objects.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+================================
+
+This document describes access to shared objects in a multithreaded scenario, whether or not is properly synchronized.
+
+Document Structure
+------------------
+
+Each component in this document has its own diagram describing all members, and whether or not access to them is synchronized or not:
+
+* **Synchronized**
```
Comment (igor-egorov) [inline docs/source/architecture/shared_objects.rst]: The cell background color manipulation was used for highlighting in the original doc.
I believe that making the font bold would not bring enough distinction to viewers.

Comment (LiraLemur) [reply docs/source/architecture/shared_objects.rst]: Done. Please check if this is ok?

**Comment by igor-egorov** [comment]
```
      span.clr { background:#C5E1F8 !important ;
      padding: 4px;
      }
      tr td span.clr  {background:#C5E1F8 !important ;
      display: block;
      margin: -10px -16px -10px -16px;
      padding: 10px;
      padding-left: 16px;
      }
```
image

File: docs/source/architecture/shared_objects.rst [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+| Block Query                     |
++---------------------------------+
+
+Synchronizer
+------------
+
++---------------------------------+
+| Synchronizer                    | 
++=================================+
+| Chain Validator.                |
```
Comment (igor-egorov) [inline docs/source/architecture/shared_objects.rst]:
```suggestion
| Chain Validator                 | 
```

**Pull request merged**